docs/user/interactive_environments.rst

```diff
@@ -0,0 +1,378 @@
+you can release the resources you were consuming. You can do this from the Environments
+tab on the Renkulab UI.
+
+Sometimes an environment will get stuck pending launch (e.g. because you requested
+GPUs and they are not yet available). In this case you can view the status and
+logs to see if there's a useful message. In the case of pending GPUs, you can
+safely wait for the resources to become available. However in other cases, more
+commonly when you are trying to customize your environment, you might notice an
+issue via the logs, and want to stop the launch so that you can fix the problem
+and try again. However, for these stuck notebooks it is not yet possible.
+You can reach out to us on `discourse <https://renku.discourse.group>`_ in this
+case.
+
+Reproducibility
+---------------
+
+What are the limitations of the Interactive Environments?
+---------------------------------------------------------
+
+
+How can I customize my Interactive Environment?
+-----------------------------------------------
+
+The launch is enabled by the content in the following files in your project:
+
+* ``Dockerfile``: defines the type of interactive environment and other software
+  installed in the environment, including the ``renku`` command-line installation.
+* ``.gitlab-ci.yml``: defines the docker build of the image based on the project's
+  ``Dockerfile``.
+
+
+
+Modifying the template's Dockerfile
+-----------------------------------
+
+The `Dockerfile` in the project is what defines the environment. In the template
+provided, the `Dockerfile`
+
+Renku Project uses Docker for containerization, which While we
+have a set of defaults that we build into a minimal Python and R images
+image, there are several reasons why you might want to build on top of these or
+write your own entirely.
+
+Dockerfile structure
+~~~~~~~~~~~~~~~~~~~~
+
+The project's ``Dockerfile`` lives in the top level of the project directory. In
+the default ``Dockerfile`` provided in the template, the first line is a ``FROM``
+statement that specifies a `versioned base docker image <https://github.com/SwissDataScienceCenter/renku-jupyter>`_.
+We add new versions periodically, but the heart of it is the set of installations
+of jupyterlab/rstudio, git, and renku::
+
+  FROM renku/singleuser:0.3.5-renku0.5.2
+
+  # or, for RStudio in the build
+
+  FROM renku/singleuser-r:0.3.5-renku0.5.2
+
+The next two statements install user-specified libraries from ``environment.yml``
+and ``requirements.txt``::
+
+  # install the python dependencies
+  COPY requirements.txt environment.yml /tmp/
+  RUN conda env update -q -f /tmp/environment.yml && \
+  /opt/conda/bin/pip install -r /tmp/requirements.txt && \
+  conda clean -y --all && \
+  conda env export -n "root"
+
+You can add to this ``Dockerfile`` in any way you'd like.
+
+.. _docker_dev:
+
+Dockerfile development
+~~~~~~~~~~~~~~~~~~~~~~
+
+If you're going to be making simple modifications to the ``Dockerfile`` (i.e. changing
```

Comment:
Maybe some Docker references? I think the official [Dockerfile reference](https://docs.docker.com/engine/reference/builder/) is quite ok.